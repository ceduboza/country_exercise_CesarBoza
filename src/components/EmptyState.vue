<template>
    <div class="empty-state">
        <q-icon :name="icon" size="10rem" />
        <span>{{ title }}</span>
    </div>
</template>

<script setup lang="ts">
defineProps({
    title: {
        type: String,
        required: true
    },
    icon: {
        type: String,
        required: true
    }
})
</script>

<style lang="scss">
.empty-state {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100%;
    padding-top: 30%;
}
</style>
