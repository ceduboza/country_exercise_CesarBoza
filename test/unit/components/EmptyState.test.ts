import { describe, it, expect } from "vitest";
import { shallowMount } from "@vue/test-utils";
import EmptyState from "../../../src/components/EmptyState.vue";

describe("EmptyState", () => {
    let wrapper

    const findIcon = wrapper => wrapper.findComponent({ name: "q-icon" });
    const findTitle = wrapper => wrapper.find("span.empty-state__title");

    const factory = ({ props = {} } = {}) => {
        wrapper = shallowMount(EmptyState, {
            props,
            global: {
                stubs: {
                    QIcon: true,
                },
            },
        });
    }

    it("should render the empty state", () => {
        const props = {
            title: "Empty State",
            icon: "empty",
        }
        factory({ props })

        const title = findTitle(wrapper)
        const icon = findIcon(wrapper)

        expect(wrapper.html()).toMatchSnapshot();

        expect(icon.exists()).toBeTruthy();
        expect(icon.attributes("name")).toBe("empty");
        expect(title.exists()).toBeTruthy();
        expect(title.text()).toBe("Empty State");
    });
});
