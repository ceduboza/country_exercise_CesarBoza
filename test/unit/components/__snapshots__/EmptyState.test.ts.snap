// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyState > should render the empty state 1`] = `
"<div class="empty-state">
  <q-icon-stub name="empty" size="10rem" tag="i" left="false" right="false"></q-icon-stub><span class="empty-state__title">Empty State</span>
</div>"
`;
